bip-cc.md

```diff
@@ -0,0 +1,207 @@
+users, increasing the cost to independently fully validate the consensus rules. In addition they can
+be used by miners to attack their competition, creating perverse incentives, centralization
+pressures and reduced network security.
+
+In computing a block's Merkle root, a 64 bytes transaction can be interpreted as an intermediate
+node in the tree in addition to a leaf. This makes it possible to fake inclusion proofs by
+pretending a 64 bytes block transaction is an inner node, as well as to pretend the inner nodes on
+one level of the tree are the actual block transactions.
+
+Since [bip-0034][BIP34] activation, explicit [bip-0030][BIP30] validation is not necessary until
+block height 1,983,702[^0].  Mandating new coinbase transactions be different from the early
+[bip-0034][BIP34] violations makes it possible to get rid of [bip-0030][BIP30] validation forever.
+Besides its unnecessary cost, another downside of [bip-0030][BIP30] validation is that it cannot be
+performed by Utreexo clients. Finally, leveraging the coinbase transaction's `nLockTime` field
+allows applications to recover the block height corresponding to a coinbase transaction without
+having to parse Bitcoin Script.
+
+# Specification
+
+For all blocks after activation the following new rules apply.
+
+Given a block at height `N`:
+- if `N % 2016` is equal to `0` the `nTime` field of the block must be set to a value higher than or
+  equal to the value of the `nTime` field of block at height `N-1` minus `7200`;
+- if `N % 2016` is equal to `2015` the `nTime` field of the block must be set to a value higher than
+  or equal to the value of the `nTime` field of block at height `N-2015`.
+
+A limit is set on the number of potentially executed signature operations in validating a
+transaction. It applies to all transactions in the block except the coinbase transaction[^1]. For
+each input in the transaction, count the number of `CHECKSIG` and `CHECKMULTISIG` in the input
+scriptSig and previous output's scriptPubKey, including the P2SH redeemScript. If the total is
+strictly higher than `2500`, the transaction is invalid. The accounting is the same as for
+[bip-0016][BIP16 specs]: a `CHECKSIG` operation accounts for `1` and a `CHECKMULTISIG` accounts for
+the number of public keys associated, or `20` if the number of public keys is greater than `16`. A
+`CHECKMULTISIG` not directly preceded by a minimally-pushed number between `1` and `16` (included)
+accounts for `20`.
+
+Transactions whose (witness-stripped) serialized size is exactly 64 bytes are made invalid.
+
+The coinbase transaction's `nLockTime` field must be set to the height of the block minus `1`[^2]
+and its `nSequence` field must not be equal to `0xffffffff`.
+
+# Rationale
+
+The restrictions on the timestamp of the first and last blocks of a difficulty adjustment period fix
+the timewarp and Murch-Zawy vulnerabilities[^3]. The latter poses mostly theoretical concerns but is
```

Comment:
```suggestion
the timewarp and Murch–Zawy vulnerabilities[^3]. The latter poses mostly theoretical concerns but is
```

Reply:
I remember hearing English had different dashes, but i didn't expect it'd come up as a Pull Request review comment one day! I'm not even sure how to produce those, and to be honest this is probably not the best use of our time.

Reply:
Yes, the en dash (–) is commonly used to connect names in a compound name to distinguish them from a double-barrelled name which uses a hyphen (-). That's how you know, for example, that the Gell-Mann–Okubo mass formula was named after two people, Gell-Mann and Okubo.

If you're not sure how to produce an en dash, copypasting it is fine too 😉  Alternatively, if neither Murch nor Zawy care about not being mistaken for the same person named Murch-Zawy, feel free to leave as is.

Reply:
I for one have never seen @murchandamus and @zawy12 in the same room at the same time!